aws/resource_aws_appstream_stack_test.go

```diff
@@ -0,0 +1,346 @@
+					resource.TestCheckResourceAttr(resourceName, "name_prefix", "terraform-"),
+					resource.TestCheckResourceAttr(resourceName, "description", description),
+				),
+			},
+			{
+				Config: testAccAwsAppStreamStackConfigWithTags(descriptionUpdated),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppStreamStackExists(resourceName, &stackOutput),
+					testAccCheckResourceAttrRfc3339(resourceName, "created_time"),
+					naming.TestCheckResourceAttrNameGenerated(resourceName, "name"),
+					resource.TestCheckResourceAttr(resourceName, "name_prefix", "terraform-"),
+					resource.TestCheckResourceAttr(resourceName, "description", descriptionUpdated),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Key", "value"),
+					resource.TestCheckResourceAttr(resourceName, "tags_all.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags_all.Key", "value"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsAppStreamStackExists(resourceName string, appStreamStack *appstream.Stack) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).appstreamconn
+		resp, err := conn.DescribeStacks(&appstream.DescribeStacksInput{Names: []*string{aws.String(rs.Primary.ID)}})
+
+		if err != nil {
+			return err
+		}
+
+		if resp == nil && len(resp.Stacks) == 0 {
+			return fmt.Errorf("appstream stack %q does not exist", rs.Primary.ID)
+		}
+
+		*appStreamStack = *resp.Stacks[0]
+
+		return nil
+	}
+}
+
+func testAccCheckAwsAppStreamStackDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).appstreamconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_appstream_stack" {
+			continue
+		}
+
+		resp, err := conn.DescribeStacks(&appstream.DescribeStacksInput{Names: []*string{aws.String(rs.Primary.ID)}})
+
+		if tfawserr.ErrCodeEquals(err, appstream.ErrCodeResourceNotFoundException) {
+			continue
+		}
+
+		if err != nil {
+			return err
+		}
+
+		if resp != nil && len(resp.Stacks) > 0 {
+			return fmt.Errorf("appstream stack %q still exists", rs.Primary.ID)
+		}
+	}
+
+	return nil
+
+}
+
+func testAccAwsAppStreamStackConfigNameGenerated() string {
+	return `
+resource "aws_appstream_stack" "test" {}
+`
+}
+
+func testAccAwsAppStreamStackConfigNamePrefix(stackName string) string {
+	return fmt.Sprintf(`
+resource "aws_appstream_stack" "test" {
+  name_prefix = %[1]q
+}
+`, stackName)
+}
+
+func testAccAwsAppStreamStackConfigComplete(description string) string {
+	return fmt.Sprintf(`
+resource "aws_appstream_stack" "test" {
+  description = %[1]q
+  storage_connectors {
+    connector_type = "HOMEFOLDERS"
+  }
```

Comment:
```suggestion

  storage_connectors {
    connector_type = "HOMEFOLDERS"
  }

```

Best to have 1 blank like before and after configuration blocks unless it's the last one.